libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -39,6 +39,42 @@ impl<'a> Validator<'a> {
         for model in schema.models() {
             let ast_model = ast.find_model(&model.name).expect(STATE_ERROR);
 
+            //doing this here for now since I want to have all field names already generated
+            // it might be possible to move this
+            if let Some(pk) = &model.primary_key {
```

Comment:
Yep that can move to the phase after attribute validation in ParserDatabase when it exists.